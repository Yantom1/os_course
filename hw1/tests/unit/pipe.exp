smash> my_sleep
random1.txt
ransmash> look here
smash> /bin/bash: not_real_command: command not found
smash> smash> smash> smash> my_sleep
random1.txt
random2.txt
sub1
sub2
sub3
smash> 0
2
Done: 3
smash> smash pid is \d+
smash> smash pid is \d+
smash> /tmp/smash_test
smash> smash> smash> \[1\] sleep 20& : \d+ \d+ secs
\[2\] sleep 10& : \d+ \d+ secs
smash> signal number 9 was sent to pid \d+
smash> signal number 9 was sent to pid \d+
smash> smash> smash> signal number 9 was sent to pid \d+
smash> smash> r directory
smash> r directory
smash> PWD not set
smash> y arguments
smash> s not exist
smash> d arguments
smash> d arguments
smash> d not found
smash> smash> smash: sending SIGKILL signal to 0 jobs:
