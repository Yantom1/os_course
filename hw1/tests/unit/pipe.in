ls | bash -c "head -c 24"
echo dont look at me | echo look here
# We still should see the stderr output
not_real_command | sleep 1
!time.sleep(3)
sleep 1 | sleep 3
!time.sleep(3)
#Verify both sleeps have died
ps | grep sleep
ls | sleep 1
!time.sleep(1)
sleep 1 | ls
!time.sleep(1)
./my_sleep 3 | grep --line-buffered --color=never ".*"
!time.sleep(3)
# Check internal commands redirection
showpid | grep smash
showpid | showpid
pwd | grep /
sleep 20&
sleep 10&
jobs | grep sleep
kill -9 1 | grep sent
kill -9 2 | grep sent
jobs | grep sleep
sleep 10 &
sleep 1 | kill -9 1
!time.sleep(1)
jobs
#
# stderr
#
# verify that we transfer the stderr by truncateing it
cat not_a_realy_path |& tail -c 12
cd not_a_realy_path |& tail -c 12
cd - |& tail -c 12
cd asdasd asdasd |& tail -c 12
kill -9 30 |& tail -c 12
kill |& tail -c 12
kill asda 1|& tail -c 12
not_real_command|& tail -c 12
jobs
quit kill
