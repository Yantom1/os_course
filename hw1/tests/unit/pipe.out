smash> my_sleep
random1.txt
ransmash> look here
smash> /bin/bash: not_real_command: command not found
smash> smash> smash> smash> my_sleep
random1.txt
random2.txt
sub1
sub2
sub3
smash> 0
2
Done: 3
smash> smash pid is 45126
smash> smash pid is 45126
smash> /tmp/smash_test
smash> smash> smash> [1] sleep 20& : 45180 0 secs
[2] sleep 10& : 45181 0 secs
smash> signal number 9 was sent to pid 45180
smash> signal number 9 was sent to pid 45181
smash> smash> smash> signal number 9 was sent to pid 45196
smash> smash> r directory
smash> r directory
smash> PWD not set
smash> y arguments
smash> s not exist
smash> d arguments
smash> d arguments
smash> d not found
smash> smash> smash: sending SIGKILL signal to 0 jobs:
